[Feat] Redis 토큰 관리 구현 완료

supsup-hae opened the pull request:
## 💡 작업 내용
- [x] Redis 개발 환경 설정
- [x] Token 관리를 Redis로 위임
- [x] AccessToken의 Blacklist 관리를 위한 Redis 적용

## 💡 자세한 설명
#### Redis 환경 구성
   - Redis 서버 설정 및 연동
   - Redis Template 구성
   - 토큰 저장소 구현

#### Token 관리 시스템 개선
   - Token 저장소를 Redis로 이관
   - 토큰 만료 처리 로직 구현
   - 토큰 검증 시스템 개선

#### Blacklist 시스템 구현
   - 로그아웃된 RefreshToken Blacklist 관리
   - 토큰 무효화 처리 로직 구현
   - Blacklist 만료 정책 설정

## 📗 참고 자료
- Redis 공식 문서
- Spring Data Redis 가이드
- JWT Blacklist 구현 레퍼런스

## 📢 리뷰 요구 사항

## ✅ 셀프 체크리스트
- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요?
- [x] Assignees, Reviewers, Labels 를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

supsup-hae committed:
:heavy_plus_sign: chore: redis 의존성 추가 (#52)

supsup-hae committed:
:memo: chore: redis 기본 속성 값 구성 (#52)

supsup-hae committed:
:memo: chore: redis 설정 파일 추가 (#52)

supsup-hae committed:
:sparkles: feat: Redis를 이용한 토큰 관리 시스템 구현 (#52)
- RefreshToken 관리를 Redis로 위임
- BlackList를 이용한 로그아웃 및 만료 토큰 관리

supsup-hae committed:
:memo: chore: Security 설정 변경 (#52)
- 주석 추가
- JwtAuthorizationFilter OAuth2FailureHandler 필터 추가
- 화이트리스트 중복 요소 제거

supsup-hae committed:
:recycle: refactor: 토큰 관리 Redis 위임 및 불필요 코드 제거 (#52)
- Token 엔티티 관련 코드 제거
- Token 관련 서비스 레이어 제거
- Token 레포지토리 제거
- User 엔티티와의 연관관계 제거
- Redis 기반 토큰 관리 로직으로 전환

supsup-hae committed:
:recycle: refactor: Auth API Swagger 문서 추가 및 네이밍 변경 (#52)

supsup-hae committed:
:recycle: refactor: User API Swagger 문서 추가 및 서비스 로직 수정 (#52)
- 기존 AccessToken을 이용하던 방식에서 인증 정보를 사용하도록 변경

supsup-hae committed:
:bug: fix: 로그인 세션 미확인 에러 해결 (#52)
- PrincipalDetails -> UserDetails 변경
- expression = user 설정

supsup-hae committed:
:memo: docs: 로깅 추가 및 반환값 변경 (#52)

supsup-hae committed:
:memo: docs: redis 설정 적용 (#52)

supsup-hae committed:
:recycle: refactor: token 만료 지정 방식 변경 (#52)

supsup-hae committed:
:recycle: refactor: User Role 기본값 변경 (#52)

Jeonkyujin reviewed (approved):
고생하셨습니다!! 덕분에 Redis를 활용한 토큰 관리의 이점을 잘 배울 수 있었습니다. 바로 merge 하시면 될 것 같습니다~

supsup-hae committed:
:sparkles: feat: Swagger API 문서 개선 및 토큰 관련 스펙 수정 (#52)

- RefreshToken 쿠키 파라미터 검증 로직 수정
- 토큰 재발급/로그아웃 API 응답 예시 추가
- API 응답 스펙 어노테이션 분리 및 재사용성 개선
